### DISCUSSION TITLE
Error TS2374: Duplicate index signature for type 'number'.

### DISCUSSION THREAD

**Discussion by smadarapu01** [pnpm/pnpm#7559] [Q&A]
we just recently updated to pnpm from yarn, while compiling application, seeing **Error TS2374: Duplicate index signature for type 'number'.** errors. We use typescript, 
Tried with clearing cache still seeing issue can you pls help on this.  brief details from log
**symlick** 
/node_modules/.pnpm/@types+jquery@3.5.22/node_modules/@types/jquery/JQuery.d.ts(13361,5): error TS2374: Duplicate index signature 
for type 'number'.
node_modules/.pnpm/@types+jquery@3.5.22/node_modules/@types/jquery/legacy.d.ts(201,5): error TS2374: Duplicate index signature for type 'string'.
**_root folder_**
/node_modules/@types/jquery/jquery.d.ts(13361,5): error TS2374: Duplicate index signature for type 'number'.
/node_modules/@types/jquery/legacy.d.ts(201,5): error TS2374: Duplicate index signature for type 'string'.

**Comment by RexSkz** [comment]
This TypeScript error occurs when there are two fields with the same type of keys and values in one interface:

```ts
interface Test {
  [key1: string]: string;
  [key2: string]: string;
}
```

It may be because of the "interface merging" feature in TypeScript:

```ts
// file 1
interface Test {
  [key1: string]: string;
}

// file 2
interface Test {
  [key2: string]: string;
}
```

Can you do `ctrl/cmd + click` on the interface name (`Test` in my example) and see if there are multiple definitions that form this scenario (the key names are different, but the types of keys and values are the same)?

If they come from two packages, the simplest way is to use [`pnpm.overrides`](https://pnpm.io/package_json#pnpmoverrides) to restrict the versions.
